Fix Run Instances window being too big in single-window mode for window at minimum size

arkology opened the pull request:
| Before | After |
| --- | --- | 
| ![Godot_v4 4-stable_win64_mnBkQdVpHO](https://github.com/user-attachments/assets/3315cdfa-4215-4bee-89fd-cdf08c8e5e62) | ![godot windows editor dev x86_64_GApEiQqL5H](https://github.com/user-attachments/assets/c22e578a-8833-44fb-987b-97b43336a9be) |

arkology committed:
Fix Run Instances window being too big

KoBeWi commented:
The new default size cuts some placeholders. The dialog is intentionally wide to facilitate all its content properly.
If you want to shrink it, it needs to be reorganized. E.g. the instance count can be in its own row.

arkology commented:
> The new default size cuts some placeholders.

While this is not a problem on FullHD monitor, on smartphones it happens even right now, in current `master`.

> If you want to shrink it, it needs to be reorganized. E.g. the instance count can be in its own row.

Will try.

arkology commented:
@KoBeWi I did the following changes:
- removed saving window position between popups - it does not work well in single window mode after minimizing window.
- moved "Main Run Args" and "Main Feature Tags" to new line.
- restored previous width, but increased height (as "Main ..." options are at new line now).

Please take a look, does it look better.

KoBeWi commented:
Seems fine now, though I think the instance count should be under the arguments, and possibly with a separator?
![image](https://github.com/user-attachments/assets/f85529ec-3ea2-437d-aaaa-8a6ee1bac23f)
Or maybe something like this.
![image](https://github.com/user-attachments/assets/085228f6-6d8d-4344-a61a-1b307534afdf)
Not sure tbh.

KoBeWi commented on editor/run_instances_dialog.cpp:
```diff
@@ -310,48 +310,50 @@ RunInstancesDialog::RunInstancesDialog() {
-	args_gc->set_columns(3);
-	args_gc->add_theme_constant_override("h_separation", 12 * EDSCALE);
-	main_vb->add_child(args_gc);
+	HBoxContainer *instance_vb = memnew(HBoxContainer);
```
```suggestion
	HBoxContainer *instance_hb = memnew(HBoxContainer);
```

arkology replied:
Oops 😄

arkology commented on editor/run_instances_dialog.cpp:
```diff
@@ -194,7 +194,7 @@ void RunInstancesDialog::_instance_tree_rmb(const Vector2 &p_pos, MouseButton p_
 
 void RunInstancesDialog::popup_dialog() {
-	popup_centered(Vector2(1200, 600) * EDSCALE);
+	popup_centered_clamped(Size2(1200, 700) * EDSCALE, 0.8);
```
Done

arkology commented:
Now it looks like this:

![godot windows editor dev x86_64_GApEiQqL5H](https://github.com/user-attachments/assets/0368ee0b-201d-49c9-b1ef-036c24d74141)

KoBeWi commented on editor/run_instances_dialog.cpp:
```diff
@@ -361,16 +343,32 @@ RunInstancesDialog::RunInstancesDialog() {
+	main_vb->add_child(memnew(HSeparator));
+
+	HBoxContainer *instance_hb = memnew(HBoxContainer);
+	instance_hb->set_h_size_flags(Control::SIZE_SHRINK_CENTER);
```
Or you could do
```suggestion
	instance_hb->set_alignment(BoxContainer::ALIGNMENT_CENTER);
```

Repiteo commented:
Thanks!
